### DISCUSSION TITLE
Alert Dialog

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#302] [Component suggestions]
<img width="720" alt="visuele weergave van de alert dialog component" src="https://github.com/user-attachments/assets/fd22ac27-e5fd-4806-bf75-e17256c7beea">

We willen graag meer leren over de **Alert Dialog** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, GitHub of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het gewaardeerd als je onderzoeksresultaten deelt. Is dit component getest met eindgebruikers of meegenomen in een toegankelijkheidsonderzoek? Deel dan vooral de inzichten die je hebt opgedaan!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1760098764502829)

---

## Naam

Alert Dialog

## Doel

Opent een venster waarin de gebruiker een actie moet uitvoeren om verder te gaan. De rest van de pagina is tijdelijk geblokkeerd.

## Gerelateerde componenten

[Alert](https://github.com/orgs/nl-design-system/discussions/301), [Modal Dialog](https://github.com/orgs/nl-design-system/discussions/237), [Dialog](https://github.com/orgs/nl-design-system/discussions/280)

## Onderzoek

[Dialogs and popovers seem similar. How are they different? - Hidde de Vries](https://hidde.blog/dialog-modal-popover-differences/)
Blogpost waarin Hidde de verschillen tussen Dialogs en pop-overs uiteen zet.

[Modal & Nonmodal Dialogs: When (& When Not) to Use Them - Nielsen Norman Group](https://www.nngroup.com/articles/modal-nonmodal-dialog/)
Artikel waarin wordt beschreven wanneer je Modal Dialogs wel en niet hoort te gebruiken.

[The Current State of Homepage UX – 8 Common Pitfalls & Best Practices - Baymard Institute](https://arc.net/l/quote/uvpyqgph)
Artikel waarin het inzicht wordt gedeeld dat Modal Dialogs op de homepagina door enkele respondenten als spam wordt ervaren.

[Modals and Pop-ups design best practices (with examples) - Youtube](https://www.youtube.com/watch?v=-I5ANqGyO4A)
Video waarin Vitaly Friedman best practices van Modal Dialogs (en pop-ups) toont.

[Where to Put Focus When Opening a Modal Dialog](https://adrianroselli.com/2025/06/where-to-put-focus-when-opening-a-modal-dialog.html)
Adrian Roselli beschrijft de verschillende afwegingen die je kan maken om te besluiten waar je de focus naar toe brengt bij het openen van een Modal Dialog.

[UI Modes & Modals – Nielsen Norman Group](https://www.nngroup.com/videos/ui-modes-modals/)
Korte video waarin wordt uitgelegd hoe UI Modes & Modals tijdelijk de interactie met de rest van de interface beperken. Benadrukt dat modals effectief kunnen zijn voor focus, maar ook snel de flow van de gebruiker kunnen verstoren als ze onnodig worden ingezet.

[Alert Fatigue in User Interfaces – Nielsen Norman Group](https://www.nngroup.com/videos/alert-fatigue-user-interfaces/)
Video over hoe te veel waarschuwingen, modals en pop-ups leiden tot 'alert fatigue', waardoor gebruikers belangrijke meldingen gaan negeren. Adviseert terughoudendheid in het gebruik van modals en duidelijke hierarchie in meldingstypes.

[Popup Problems – Nielsen Norman Group](https://www.nngroup.com/videos/popup-problems/)
Korte video over veelvoorkomende problemen met pop-ups: ze onderbreken de gebruiker, worden vaak als opdringerig ervaren en kunnen toegankelijkheidsproblemen veroorzaken. Biedt suggesties om ze minder storend en relevanter te maken.

[Open Dialog – Scott O’Hara](https://www.scottohara.me/blog/2019/03/05/open-dialog.html)
Scott O’Hara gaat diep in op de toegankelijkheidsaspecten van het `<dialog>`-element in HTML. Hij bespreekt het gedrag, de valkuilen, en hoe ontwikkelaars modals correct kunnen implementeren.

[Modal Dialogue – GOV.UK Design System](https://components.publishing.service.gov.uk/component-guide/modal_dialogue)
De GOV.UK-componentgids beschrijft hoe modale dialogen binnen overheidswebsites moeten worden toegepast: enkel bij kritieke acties, altijd met toetsenbordnavigatie en een duidelijke manier om te sluiten. Richt zich sterk op toegankelijkheid en consistent gebruik.

**Comment by veslav3** [comment]
Deze is geïmplementeerd door Amsterdam: https://designsystem.amsterdam/?path=/docs/components-containers-dialog--docs

**Reply by eslook** [reply]
Dat lijkt zo, echter dat is een Modal Dialog, niet een Alert Dialog. Dit zie je aan dat je de modal altijd kan sluiten zonder een actie te ondernemen, in plaats van dat je een actie moet ondernemen voordat je de dialog mag sluiten.

**Comment by eslook** [comment]
## Utrecht



- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-alert-dialog--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/alert-dialog)

---

### Screenshots / Varianten

**Storybook**

Info (Default)
<img width="815" height="137" alt="Screenshot 2025-10-10 at 15 47 48" src="https://github.com/user-attachments/assets/fff87757-318a-453f-bee7-b2f18e7749bc" />

Warning
<img width="815" height="137" alt="Screenshot 2025-10-10 at 15 47 53" src="https://github.com/user-attachments/assets/7f4c647a-c0a4-48e4-a672-c05fad347461" />

Error
<img width="815" height="137" alt="Screenshot 2025-10-10 at 15 47 55" src="https://github.com/user-attachments/assets/db59a0bd-05e3-4e1c-aea7-704970496e06" />

---

### Notities

- Onder de naam: Alert Dialog.
- Bestaat uit content en een optionele icon.
- Bevat drie varianten (error, info, warning), waarvan info de default is.
- Responsive gedrag: geen verandering.
- Richtlijnen beschikbaar in de README middels GitHub en Storybook.

**Comment by JJGrootveld** [comment]
## KVK

### Screenshots / Varianten
#### Storybook
<img width="2082" height="484" alt="CleanShot 2025-10-10 at 16 24 14@2x" src="https://github.com/user-attachments/assets/7ab41d81-5b9f-4fab-a658-e148ebc32ccd" />

<img width="2126" height="2154" alt="CleanShot 2025-10-10 at 16 23 56@2x" src="https://github.com/user-attachments/assets/8d0bf231-c487-4ed3-9551-439375aebbd0" />


### Documentatie in Figma

<img width="2404" height="1658" alt="CleanShot 2025-10-10 at 16 29 06@2x" src="https://github.com/user-attachments/assets/c7e3174d-17be-4636-ae30-b2772aff8077" />

<img width="2352" height="842" alt="CleanShot 2025-10-10 at 16 29 21@2x" src="https://github.com/user-attachments/assets/faf1299d-dd45-416f-9cc4-60629d0cf3b5" />

<img width="2412" height="1760" alt="CleanShot 2025-10-10 at 16 29 31@2x" src="https://github.com/user-attachments/assets/56459659-11ee-4002-9b1a-a53b6e7e96fc" />

<img width="2404" height="1068" alt="CleanShot 2025-10-10 at 16 29 37@2x" src="https://github.com/user-attachments/assets/77a5e594-9e1d-4667-bec8-ef8f00b5169b" />

### Component in Figma
<img width="3070" height="1066" alt="CleanShot 2025-10-10 at 16 31 04@2x" src="https://github.com/user-attachments/assets/68f85dc7-0b12-4553-8b4d-7f32461cb2b0" />

### Onderzoek naar dit component
- De blauwe "Info" variant van het Notification component wordt in een aantal van onze onderzoeken makkelijk over het hoofd gezien **tenzij** zij echt bewust de pagina scannen en op zoek zijn naar informatie.

**Reply by eslook** [reply]
@JJGrootveld ik vraag me af of dit meer een [Alert](https://github.com/orgs/nl-design-system/discussions/301) is dan een Alert Dialog? De Alert (ook bekend als Message, Notification) is een component dat bijvoorbeeld inline gebruikt word, of in een deel van het scherm verschijnt zonder de interactie met de onderliggende pagina te blokkeren.

Ik denk ook niet dat het een [Modal Dialog](https://github.com/orgs/nl-design-system/discussions/237) is of een [Non Modal Dialog](https://github.com/orgs/nl-design-system/discussions/280) is, omdat de enige interactieve elementen het kruisje zijn en tekst links. Dit komt meer over met de Alert, in vergelijking met deze twee die het kruisje, tekst links, en de interactieve buttons in de footer heeft.

Hoe zie jij dit?

**Reply by JJGrootveld** [reply]
Je hebt gelijk! Dit is meer een notification/alert component omdat er geen directe actie aan verbonden is. Ik had de vergelijking alleen visueel gemaakt; maar ik denk toch relevant omdat de toevoeging van de acties binnen ons component veel wordt aangevraagd. Dus ik ben ook benieuwd of we een pivot moeten maken naar een alert dialog, maar dan wordt de context groter.

Op dit moment is het uitsluitend een informatief component richting de gebruiker, maar met acties en de alert dialog zoals hij hier wordt ingestoken: "Opent een venster waarin de gebruiker een actie moet uitvoeren om verder te gaan. De rest van de pagina is tijdelijk geblokkeerd."; is dan al eigenlijk fundamenteel anders. En dan vinden jullie dus ook aangezien het een apart component voorstel is.

Ik ben zelf vooral heel benieuwd hoe DS gebruikers (vooral ook in design/Figma) de juiste keuze gaan maken tussen de alert en alert dialog. Dat is mijn terughoudendheid bij het splitsen van de componenten.

**Comment by eslook** [comment]
## DSO



- [Documentatie website](https://www.dso-toolkit.nl/)
- [Storybook via documentatie website](https://www.dso-toolkit.nl/79.0.1/components/modal/)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit)

---

### Screenshots / Varianten

**Storybook**

Confirm
![Scherm­afbeelding 2024-04-08 om 08 46 22](https://github.com/nl-design-system/backlog/assets/545207/eb121aad-51ca-4c7f-93c1-68d6a8bdb386)

Small viewport

![Scherm­afbeelding 2024-04-08 om 08 49 49](https://github.com/nl-design-system/backlog/assets/545207/28862479-65c5-4a2f-ad05-4260f40d44f1)

Passive
![Scherm­afbeelding 2024-04-08 om 08 46 14](https://github.com/nl-design-system/backlog/assets/545207/e16e498c-adf2-4187-82e8-e63d575a6805)

Active
![Scherm­afbeelding 2024-04-08 om 08 46 31](https://github.com/nl-design-system/backlog/assets/545207/720744be-991b-45be-b86b-6dae8a8381d0)

Loading
![Scherm­afbeelding 2024-04-08 om 08 46 38](https://github.com/nl-design-system/backlog/assets/545207/cc51030d-fb26-4340-8a50-f6da58e3d49a)

With Datepicker
![Scherm­afbeelding 2024-04-08 om 08 46 46](https://github.com/nl-design-system/backlog/assets/545207/c6674ea5-746b-49f5-8217-19451c3b4388)

Fullscreen
![Scherm­afbeelding 2024-04-08 om 08 46 54](https://github.com/nl-design-system/backlog/assets/545207/0c44ea36-03fd-4e7d-84df-f767c85f06d9)

---

### Notities

- Bestaat onder de naam Modal. Kan toegepast worden zowel als Alert Dialog als Modal Dialog.
- Richtlijnen beschikbaar. Inclusief richtlijnen over toegankelijkheid en role (dialog vs. alert dialog).
- Veel verschillende vormen in Storybook.
- Zie ook dezelfde [comment](https://github.com/orgs/nl-design-system/discussions/237#discussioncomment-9022610) in Modal Dialog.

**Comment by eslook** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/modal)

---

### Screenshots / Varianten

**Documentatie website**

Modal met backdrop
Deze responsive modal wordt doorgaans het meest gebruikt voor extra informatie en korte gebruikersacties. De backdrop zorgt ervoor dat de focus op de modal komt te liggen. De gebruiker wordt niet afgeleid door informatie achter de modal.

![Scherm­afbeelding 2024-04-08 om 08 57 22](https://github.com/nl-design-system/backlog/assets/545207/9c7aa953-92f7-4c61-ba7e-45c4ca240253)

![Scherm­afbeelding 2024-04-08 om 08 57 37](https://github.com/nl-design-system/backlog/assets/545207/76b846c1-2003-4a9a-87b0-974bd1b867ab)

Modal voor afbeeldingen
Deze modal laat een afbeelding vergroot zien.
![Scherm­afbeelding 2024-04-08 om 08 57 51](https://github.com/nl-design-system/backlog/assets/545207/499305ba-9f73-4522-8a49-cd313bb5ffa3)

XL modal
![Scherm­afbeelding 2024-04-08 om 08 58 00](https://github.com/nl-design-system/backlog/assets/545207/770f251b-d38a-46fe-aa4e-312406c9a4bc)

Large modal
![Scherm­afbeelding 2024-04-08 om 08 58 14](https://github.com/nl-design-system/backlog/assets/545207/2d7fc0d9-2b2c-463f-b2c1-7554b4c7d97c)

Small modal
![Scherm­afbeelding 2024-04-08 om 08 58 23](https://github.com/nl-design-system/backlog/assets/545207/d15f7e3b-916c-4470-9289-35dc3e2af648)

---

### Notities
- Bestaat onder de naam Modal. Kan toegepast worden zowel als Alert Dialog als Modal Dialog.
- Richtlijnen beschikbaar.
- Naast breedte ook classes voor hoogte beschikbaar.
- Button in principe links uitgelijnd, maar lijkt rechts gepositioneerd te worden wanneer footer tekst bevat.
- Sluit button met label.
- Zie ook dezelfde [comment](https://github.com/orgs/nl-design-system/discussions/237#discussioncomment-9022610) in Modal Dialog.

**Comment by eslook** [comment]
Tijdens de estafettemodeldag van vrijdag 28 november hebben we een werksessie gedaan over de Alert Dialog. Hier vind je de [FigJam](https://www.figma.com/board/RmHN8rnQANGz0NYg6fRIab/NL-Design-System---Estafettemodeldag?node-id=0-1).

## Context

Het verschil tussen de Modal Dialog en de Alert Dialog is dat je met de Alert Dialog verplicht bent om een actie te ondernemen voordat je verder gaat (bijvoorbeeld annuleren of accepteren), terwijl je met de Modal Dialog ook zonder actie te ondernemen de modal kan sluiten (middels bijvoorbeeld escape of de sluit knop).

De Modal Dialog is op dit moment al Community. De Alert Dialog is onderweg naar Community. Maar hier is de vraag naar voren gekomen:

“Moet Alert Dialog wel een los component zijn van Modal Dialog, of is Alert Dialog een implementatie van Modal Dialog?”

Originele discussie op Slack:
https://codefornl.slack.com/archives/C7KDQRXMZ/p1760687089821019

## Activiteiten

- Klassikaal verduidelijken wat de verschillen zijn tussen de gerelateerde componenten: Alert Dialog, Modal Dialog, Non-Modal Dialog, Alert, Note (Spotlight Section) ter voorbereiding op de sessie.
- Individueel stickies plaatsen over hoe jij de toekomst van de Alert Dialog ziet:
  - Alert Dialog en Modal Dialog moeten 2 aparte componenten zijn, want...
  - Alert Dialog moet een implementatie zijn van Modal Dialog, want...
  - Ik heb een andere gedachte, namelijk...
- Klassikaal de stickiesbespreken om discussie op te starten.
- Individueel stickies plaatsen over welke verschillende varianten of stories van de Alert Dialog en Modal Dialog jij wilt zien.
- Klassikaal de stickies bespreken.
  - Allereerst om scenario's uit te werken en te beslissen welk component deze toe doet.
  - Vervolgens enkel om de kleur aan te passen om te weergeven bij welk component deze hoort.
  - De stickies uitbreiden aan de hand van discussies.
  - Vraagteken stickers geplaatst waar nog onderzoek voor gewenst is.

Dit zorgde geleidelijk voor duidelijke punten.

## Wegneempunten

### 1. Alert Dialog vs Modal Dialog

**Stelling:**
De keuze voor de een of de ander kan besloten worden met de vraag: is de dialog dismissable? Zo ja, dan is het een Modal Dialog. Zo nee, dan is het een Alert Dialog. Hoe beslis je als UX designer of de dialog dismissable is? Word de dialog getriggerd door de gebruiker, is het onderdeel van de user flow, dan is het een Modal Dialog. Word de dialog getriggered door het systeem, bijvoorbeeld aan de hand van de sessie, dan is het een Alert Dialog.

> Dit is bewezen tijdens de werksessie “varianten en stories sorteren naar Alert Dialog vs Modal Dialog”

Met deze stelling gaan wij verder. Deze stelling kan op Community niveau door de Community aan de tand gesteld worden.

### 2. Verbeteren van de documentatie

- We willen de omschrijvingen van de gerelateerde componenten verbeteren -> https://github.com/nl-design-system/documentatie/issues/3163.
- We willen een artikel over hoe je kiest tussen de twee componenten -> https://github.com/nl-design-system/documentatie/issues/3162.

### 3. Potentieel onderzoek

- Meerdere dialogs tegelijk zie [sticky](https://www.figma.com/board/RmHN8rnQANGz0NYg6fRIab/Estafettemodeldag-28-november-2025?node-id=38-168&t=Gqf7UboFFCzlNrmH-4)  
- Image Light Box [zie sticky](https://www.figma.com/board/RmHN8rnQANGz0NYg6fRIab/Estafettemodeldag-28-november-2025?node-id=46-185&t=Gqf7UboFFCzlNrmH-4) 
- Gedrag na login timeout [zie sticky](https://www.figma.com/board/RmHN8rnQANGz0NYg6fRIab/Estafettemodeldag-28-november-2025?node-id=9-53&t=Gqf7UboFFCzlNrmH-4)
- Gewenst gedrag voor toetsenbord gebruikers en screenreader gebruikers [zie sticky](https://www.figma.com/board/RmHN8rnQANGz0NYg6fRIab/Estafettemodeldag-28-november-2025?node-id=5-2306&t=Gqf7UboFFCzlNrmH-4) 
- Formulier in (alert) dialog [zie sticky](https://www.figma.com/board/RmHN8rnQANGz0NYg6fRIab/Estafettemodeldag-28-november-2025?node-id=15-241&t=Gqf7UboFFCzlNrmH-4) 
- Links in alert dialog [zie sticky](https://www.figma.com/board/RmHN8rnQANGz0NYg6fRIab/Estafettemodeldag-28-november-2025?node-id=58-1146&t=Gqf7UboFFCzlNrmH-4) 
- Alert dialog met `<details>` [zie sticky](https://www.figma.com/board/RmHN8rnQANGz0NYg6fRIab/Estafettemodeldag-28-november-2025?node-id=86-1209&t=Gqf7UboFFCzlNrmH-4) 
- Stelling over dismissable [zie sticky](https://www.figma.com/board/RmHN8rnQANGz0NYg6fRIab/Estafettemodeldag-28-november-2025?node-id=5-2333&t=Gqf7UboFFCzlNrmH-4) 

### 4. Keuze voor enkel component versus twee componenten

- Wij kiezen voor de duidelijke scheiding tussen de twee componenten.
- Afwijkingen zijn mogelijk, en toegestaan, want het is Help Wanted.
- Aan de hand van de Community implementaties kunnen we leren hoe dit uiteindelijk uitpakt in Candidate. Het is ook goed mogelijk dat dit uiteindelijk betekent dat Modal Dialog en Alert Dialog worden samengevoegd naar Modal Dialog in Candidate, als ervaring uit de Community blijkt.
- We hebben al getest in deze werksessie met varianten en stories (use cases / situaties). Meer zullen vanuit de Community moeten komen tijdens de Community fase.

De scheiding van deze twee componenten komt niet vaak voor in andere libraries. Echter NL Design System legt een focus op toegankelijkheid, waardoor er een expliciet verschil is tussen de twee componenten. Dit is ook in lijn met hoe NL Design System de verschillende inputs opsplitst: Text Input, Password Input, Date Input, File Input, Number Input, etc.

### 5. Gerelateerde componenten op de website

- Voor de korte termijn, voor deze specifieke componenten, willen we op de website noteren welke componenten gerelateerd zijn. Deze relaties staan al in de GitHub backlog/discussie. Uit ervaring met de community blijkt dat het handig is om bekend te zijn met deze groep van componenten, om een goede keuze te kunnen maken voor een component -> https://github.com/nl-design-system/documentatie/issues/3160.

Voor de lange termijn is dit idealiter een betere oplossing, echter is daar op dit moment geen plan voor, ook omdat de website word herbouwd in de nabije toekomst.

**Comment by eslook** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Alert Dialog op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=alert+dialog)
